addons/mail/static/src/models/composer_view/composer_view.js

```diff
@@ -300,105 +308,43 @@ registerModel({
-                if (composer.thread.model === 'mail.channel') {
-                    Object.assign(postData, {
-                        subtype_xmlid: 'mail.mt_comment',
-                    });
-                } else {
-                    Object.assign(postData, {
-                        subtype_xmlid: composer.isLog ? 'mail.mt_note' : 'mail.mt_comment',
-                    });
-                    if (!composer.isLog) {
-                        params.context = { mail_post_autofollow: true };
-                    }
-                }
-                if (this.threadView && this.threadView.replyingToMessageView && this.threadView.thread !== this.messaging.inbox) {
-                    postData.parent_id = this.threadView.replyingToMessageView.message.id;
-                }
-                const { threadView = {} } = this;
-                const { thread: chatterThread } = this.chatter || {};
-                const { thread: threadViewThread } = threadView;
-                const messageData = await this.env.services.rpc({ route: `/mail/message/post`, params });
-                if (!this.messaging) {
-                    return;
-                }
-                const message = this.messaging.models['Message'].insert(
-                    this.messaging.models['Message'].convertData(messageData)
-                );
-                for (const threadView of message.originThread.threadViews) {
-                    // Reset auto scroll to be able to see the newly posted message.
-                    threadView.update({ hasAutoScrollOnMessageReceived: true });
-                }
-                if (chatterThread) {
-                    if (this.exists()) {
-                        this.delete();
-                    }
-                    if (chatterThread.exists()) {
-                        chatterThread.fetchData(['followers', 'suggestedRecipients']);
-                    }
-                }
-                if (threadViewThread) {
-                    if (threadViewThread === this.messaging.inbox) {
-                        if (this.exists()) {
-                            this.delete();
-                        }
-                        this.env.services['notification'].notify({
-                            message: _.str.sprintf(this.env._t(`Message posted on "%s"`), message.originThread.displayName),
-                            type: 'info',
-                        });
-                    }
-                    if (threadView && threadView.exists()) {
-                        threadView.update({ replyingToMessageView: clear() });
-                    }
-                }
-                if (composer.exists()) {
-                    composer._reset();
-                }
-            } finally {
-                if (composer.exists()) {
-                    composer.update({ isPostingMessage: false });
-                }
+            this.composer.activeThread.sender.queueMessage(composition);
+            this.composer._reset();
```

Comment:
Could be nice to properly flag method as public (i.e. no `_` prefix).

Reply:
I agree